iroh-net/src/discovery.rs

```diff
@@ -165,7 +261,19 @@ impl Discovery for ConcurrentDiscovery {
             .iter()
             .filter_map(|service| service.resolve(endpoint.clone(), node_id));
 
-        let streams = futures_buffered::Merge::from_iter(streams);
+        let streams = futures_buffered::MergeBounded::from_iter(streams);
+        Some(Box::pin(streams))
+    }
+
+    async fn subscribe(&self) -> Option<BoxStream<(NodeId, DiscoveryItem)>> {
+        let mut streams = vec![];
+        for service in self.services.iter() {
+            if let Some(stream) = service.subscribe().await {
+                streams.push(stream)
+            }
+        }
+
+        let streams = futures_buffered::MergeBounded::from_iter(streams);
```

Comment:
`MergeBounded` is bounded in the amount of concurrent streams it contains, yeah.
There is [a vague comment about this in `MergeUnbounded`](https://docs.rs/futures-buffered/latest/futures_buffered/struct.MergeUnbounded.html).

`MergeBounded` is just a small wrapper around `FuturesUnorderedBounded`, and is the thing with the limited capacity.
We're using `MergeBounded`'s `FromIterator` implementation here, which uses `FuturesUnorderedBounded::from_iter`, which [sets the capacity to the size of the iterator](https://docs.rs/futures-buffered/latest/src/futures_buffered/futures_unordered_bounded.rs.html#315).
I think this means that if you were to `streams.push` here immediately before polling the stream, that'd panic.
But because we don't do this we're fine. And if we were polling a bit before pushing (making some streams "finish" first), we'd be fine to push more work into it.

Reply:
All of this said, this is not an actual PR change. `Merge` was used before this PR, and that simply got deprecated and replaced by `MergeBounded`.